### PR TITLE
Create CI Action to verify recipes at PRs

### PR DESCRIPTION
**Pull request by linuxianer99**
This creates a Github Action that run a bitbake dry-run to verify the recipe syntax

### CODE REVIEW

**Comment by linuxianer99** [comment]
@lordfolken  is this what you have in mind ?

**Comment by lordfolken** [comment]
Yes something like that should do it.
Two things come to mind:
 * use the hcr.io/openvario/ovbuild-container
 * verify all machine types: Github has this [matrix](https://ncorti.com/blog/howto-github-actions-build-matrix) function.

File: .github/workflows/verify_p.yml [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+        uses: actions/checkout@v2
+        
+      - name: Update submodules
+        run: git submodule update --init
+      
+      # Install necessary packages
+      - name: Install build dependencies
+        run: |
+          sudo apt-get update
+          sudo apt-get install -y ffmpeg gawk wget git diffstat unzip texinfo gcc build-essential chrpath socat cpio python3 python3-pip python3-pexpect xz-utils debianutils iputils-ping python3-git python3-jinja2 libegl1-mesa libsdl1.2-dev pylint3 xterm python3-subunit mesa-common-dev zstd liblz4-tool
```
Comment (MaxKellermann) [inline .github/workflows/verify_p.yml:39]: What's the use of these packages? They are only needed to actually build XCSoar.

Comment (linuxianer99) [reply .github/workflows/verify_p.yml:39]: Yes. I am also working on a different container for building complete images, not only checking recipes. I try to minimize the variants of the workflow files. 
For only checking the recipes (--dry-run) these packages are not necessary.

**Comment by linuxianer99** [comment]
> Yes something like that should do it. Two things come to mind:
> 
> * use the hcr.io/openvario/ovbuild-container
> * verify all machine types: Github has this [matrix](https://ncorti.com/blog/howto-github-actions-build-matrix) function.

@lordfolken i will have a look on this ... but i think for the recipe check, it doesn't matter, which image or target is built ....

**Comment by linuxianer99** [comment]
@lordfolken I decided not to use the container, as it is unnecessary overhead in a defined build environment like the github actions.

Matrix build is included ..

**Pull request merged**